### PR TITLE
tkt-72659: vmware.match_datastores_with_datasets

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/vmware.py [hunk 1]
```diff
@@ -188,17 +320,10 @@ def get_datastores(self, data):
...
         connect.Disconnect(server_instance)
 
-        # Datastore names are unique among different esxi host under a single vcenter server. As we only require
-        # datastore names and additional information is not needed for datastore choices, we refine the datastores
-        # dict and send back only the datastore names
-
-        return sorted({
-            name for host in datastores.values() for name in host
-        })
+        return datastores
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/vmware.py]: My only concern is that we are changing the return type of this method, arent we?

Isnt there anything using it?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/vmware.py]: `get_datastores` was returning `[str]` and now it is the same, nothing has changed, hasn't it?

Comment (william-gr) [reply src/middlewared/middlewared/plugins/vmware.py]: Isnt datastores a dict? Before it was returning a list?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/vmware.py]: @william-gr, `get_datastores` method's implementation has been redefined above ;)

Comment (william-gr) [reply src/middlewared/middlewared/plugins/vmware.py]: Ah, I was looking at the wrong place ;)

**Pull request merged**